source/common/quic/envoy_quic_client_stream.cc

```diff
@@ -354,15 +355,16 @@ void EnvoyQuicClientStream::OnStreamReset(const quic::QuicRstStreamFrame& frame)
   quic::QuicSpdyClientStream::OnStreamReset(frame);
   ASSERT(read_side_closed());
   if (write_side_closed() && !end_stream_decoded_and_encoded) {
-    runResetCallbacks(quicRstErrorToEnvoyRemoteResetReason(frame.error_code));
+    runResetCallbacks(quicRstErrorToEnvoyRemoteResetReason(frame.error_code), absl::string_view());
   }
 }
 
 void EnvoyQuicClientStream::ResetWithError(quic::QuicResetStreamError error) {
   ENVOY_STREAM_LOG(debug, "sending reset code={}", *this, error.internal_code());
   stats_.tx_reset_.inc();
   // Upper layers expect calling resetStream() to immediately raise reset callbacks.
-  runResetCallbacks(quicRstErrorToEnvoyLocalResetReason(error.internal_code()));
+  runResetCallbacks(quicRstErrorToEnvoyLocalResetReason(error.internal_code()),
+                    absl::string_view());
```

Comment:
Is it worth using QuicResetStreamError details here?

Reply:
That's the issue I brought up in the internal bug. If the stream is reset with stream error, it's likely converted to LocalReset in quicRstErrorToEnvoyLocalResetReason and a non-empty transport failure reason will hit the ASSERT at https://github.com/envoyproxy/envoy/blob/0b093461c7ea4fa19b3ad631f048c6673e2b3f1b/source/common/router/upstream_codec_filter.h#L68

Reply:
right.  my response on the issue was you could remove that assert, rename transport_failure_reason to failure_reason and only set codec_error if failure_reason was empty.  you could do that here or in a follow-up, or just land this as a partial improvement. Thoughts?
/wait-any

Reply:
Sure I'd like to do that in a follow up